### PR TITLE
Detects if withOverride is being using without parent provider

### PR DESCRIPTION
**Pull request by rishabhpoddar**
@dulowski-marek  please feel free to merge into your branch once you have reviewed it

Related to: https://github.com/supertokens/supertokens-auth-react/pull/255

### CODE REVIEW

File: lib/ts/components/componentOverride/componentOverrideContext.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
 import { ComponentOverride } from "./componentOverride";
 
-type ContextType<T> = {
-    [K in keyof T]?: ComponentOverride<any>;
-};
+type ContextType<T> =
+    | {
+          [K in keyof T]?: ComponentOverride<any>;
+      }
+    | "IS_DEFAULT";
```
Comment (dulowski-marek) [inline lib/ts/components/componentOverride/componentOverrideContext.ts:8]: Why use `IS_DEFAULT` string instead of `undefined` or `null` for instance?

**Review by dulowski-marek — COMMENTED** [review]
Just curious about the reason

File: lib/ts/components/componentOverride/componentOverrideContext.ts (hunk 1 above)
Comment (rishabhpoddar) [reply lib/ts/components/componentOverride/componentOverrideContext.ts:8]: Just to be more explicit.. Could have used `undefined` or `null` as well.. But they we would not be able to differentiate if the user has provided `undefined` as opposed to it being our mistake where we forgot to add a parent provider.

Comment (dulowski-marek) [reply lib/ts/components/componentOverride/componentOverrideContext.ts:8]: Right, the mistake might also be related to passing invalid value in normalize config as well. Let's go this way.

**Review by dulowski-marek — APPROVED** [review]

**Pull request merged**